dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java

```diff
@@ -462,4 +465,32 @@ public boolean canDefaultPageToDefaultLanguage () {
     public boolean canDefaultFileToDefaultLanguage() {
         return Config.getBooleanProperty("DEFAULT_FILE_TO_DEFAULT_LANGUAGE",true);
     }
+
+	/**
+	 * {@inheritDoc}
+	 * @return
+	 */
+    @Override
+	public Language makeDefault(Long languageId, final User user)
+			throws DotDataException, DotSecurityException {
+    	if(!user.isAdmin()){
+    		throw new DotSecurityException("Only admin users are allowed to perform a default language switch.");
+		}
+		factory.makeDefault(languageId);
+        return factory.getDefaultLanguage();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * @return
+	 */
+	public void transferAssets(final Long oldDefaultLanguage, final Long newDefaultLanguage, final User user)
```

Comment:
WrapInTransaction
Make sure the flushCache and refreshAll runs into a commit listener

Reply:
done